mks/model/stuf_02_04.py

```diff
@@ -314,7 +320,38 @@ def extract_verbintenis_data(persoon_tree: Tag):
 
 
 def extract_address(persoon_tree: Tag, is_amsterdammer):
-    result = []
+    addresses_vbl_source = persoon_tree.find_all("PRSADRVBL")
+    addresses_cor_source = persoon_tree.find_all("PRSADRCOR")
+    addresses_vbl = format_addresses(
+        addresses_vbl_source, ADDR_VBL, is_amsterdammer)
+    addresses_cor = format_addresses(
+        addresses_cor_source, ADDR_COR, is_amsterdammer)
+    addresses = merge_sort_addresses(addresses_vbl, addresses_cor)
+
+    return addresses
+
+
+def merge_sort_addresses(*address_lists):
+    all_addresses = list(chain(*address_lists))
+    if is_nil(all_addresses):
+        return {}, []
+    all_addresses.sort(
+        key=lambda x: x["begindatumVerblijf"] or date.min, reverse=True)
+    end = all_addresses[0]["einddatumVerblijf"]
+    hasCurrent = end is None or end > date.today()
+    # Laatste inschrijfdatum
+    address_current = all_addresses[0] if hasCurrent else {}
+    # Alle andere, beginnend met het vorige adres
+    address_history = all_addresses[1:] if hasCurrent else all_addresses[0:]
```

Comment:
Dit spul dus liever in een andere functie. 

En kun je mij uitleggen in welke gevallen het nodig is om zo'n uitgebreide check te doen? 

Volgens mij kun je, zoals ik gesuggereerd had, gewoon het eerste adres nemen als current. Je sorteert op beginDatumVerblijf waarbij het adres met de laatste begin datum (dus de meest recent) als eerste in de lijst staat. In welk geval is dat niet het adres dat wij willen tonen bij "Adres" in de front-end?

Mocht er maar één adres zijn moet er idd wel even een check gedaan worden op de `address_history`. b.v:

```
address_history = []
if len(all_addresses) > 1:
    address_history = all_addresses[1:]
```